abra/data.py

```diff
@@ -368,39 +373,35 @@ def create_epochs(self, event_timestamps, conditions=None, pre_event=200, post_e
         pre_event: milliseconds before defined starting timestamps
         post_event: milliseconds after defined starting timestamps
         pupil_baseline: Baselining for pupil size data,
-                        baseline period will be the milliseconds
-                        before pre_event
+                     baseline period will be the milliseconds
+                     before pre_event
         """
 
-        #Create an empty array for storing the epoch information
+        # Create an empty array for storing the epoch information
         win_size = int((pre_event+post_event)*self.sample_rate/1000)
         all_epochs = []
-        
         # Iterate timestamp events get each epoch pupil size
         for i in range(len(event_timestamps)):
-            start = event_timestamps[i]-pre_event
-            end = event_timestamps[i]+post_event
-            idx = (self.timestamps >= (event_timestamps[i]-pre_event)) & (self.timestamps <= (event_timestamps[i]+post_event))
+            start = event_timestamps[i]-(pre_event*self.sample_rate/1000)
+            end = event_timestamps[i]+(post_event*self.sample_rate/1000)
+            idx = (self.timestamps > (start)) & (self.timestamps <= (end))
 
-            if np.sum(idx) != win_size:
+            if end-start != win_size:
                 non_zero_idx = np.nonzero(idx)
-
                 # Explicitly set all values beyond window size to False
                 # Enforcing the length to be the defined window size
                 idx[non_zero_idx[0][0]+win_size:]=False
-            epoch_pupil = self.pupil_size[idx]
```

Comment:
This is a problem that the movement data is removed. Please check to see to make sure to keep the eye movement data.